mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -66,6 +66,7 @@ public void deleteAnnotation(String key) {
     }
 
     public void insertAnnotation(MailboxAnnotation mailboxAnnotation) {
+        Preconditions.checkArgument(!mailboxAnnotation.isNil());
```

Comment:
AnnotationMapper is supposed to be a contract and should come with its testsuite. If you think that an upper layer (MailboxMapper) cover AnnotationMapper contract, it makes sense to only provide a MailboxMapper testsuite.
But, if some cases are not covered by this strategy, you have to add a testsuite on AnnotationMapper to make sure everything is covered.